content/blog/kubernetes-is-better-with-pulumi/index.md

```diff
@@ -0,0 +1,94 @@
+---
+title: "Kubernetes Is Better with Pulumi"
+date: 2020-06-02
+meta_desc: "Ten reasons why you should deploy modern apps and Kubernetes with Pulumi"
+meta_image: dangerous_pulumi.gif
+authors:
+   - levi-blackstone
+   - sophia-parafina 
+tags:
+    - kubernetes
+---
+
+You’ve containerized your application, and it’s running great on your desktop using Docker Compose or even Swarm. But now it’s time to test it locally with minikube and then put it into production with Kubernetes. Manifests are a bit like Compose files - it’s just YAML, right?
+
+<!--more-->
+
+## It’s Dangerous to Go Alone. Take this
+
+Recently, Pulumi engineer, [Levi Blackstone](https://twitter.com/levi_blackstone) posted a Twitter thread on why he thinks Pulumi is the best way to build and deploy modern applications on Kubernetes. Levi might be a bit biased, but he makes solid points for using Pulumi with Kubernetes, whether you’re a developer or operator.
+
+![Legend of Pulumipus](dangerous_pulumi.gif)
+
+## For Devs
+
+First, let’s take a look at Kubernetes from a dev’s point of view.
+
+- **Use the full Kubernetes API surface** – Each of Pulumi’s Kubernetes SDKs (TypeScript, Python, .NET and Go) are generated from the OpenAPI spec. This is critical because it means that resource definitions in any language are a 1:1 match with the upstream API. In the example below, the deployment manifest on the left corresponds to the typescript on the right.
+
+![Kubernetes API](kubernetes_api.jpg)
+
+- **Conditional logic** – Another significant drawback of YAML is the lack of conditional logic. It’s common to change parameters depending on the environment, but this usually calls for separate tools to template your YAML. With Pulumi, this logic can go right into your resource definitions. It’s not uncommon to develop your application locally using minikube, which uses ClusterIP to expose the application. When you deploy in the cloud, Kubernetes uses a LoadBalancer to manage ingress and egress. The example below shows how you can deploy to either environment without having to rewrite the manifest.
+
+```ts
+var frontendServiceType string
+if isMinikube {
+   frontendServiceType = “ClusterIP”
+} else {
+   frontendServiceType = “LoadBalancer”
+}
+
+frontendService, err := corev1.NewService(ctx, “frontent”, &corev1.ServiceArgs{
+   Metadata: &metav1.ObjectMetaArgs{
+      Labels: frontendLabels,
+      Name: pulumi.String(“frontend”),
+   },
+   Spec: &corev1.ServiceSpecArgs{
+      Type: pulumi.String(frontendServiceType),
+      Ports: corev1.ServicePortArray{
+         corev1.ServicePortArg{
+            Port: pulumi.Int(80),
+         },
+      },
+      Selector: frontendLabels,
+   },
+})
+```
+
+- **Strongly typed resources** – A tricky part about defining Kubernetes resources in YAML/JSON is the lack of typing. It's hard to track typos, and you have to apply the resource to a cluster to get any feedback. Pulumi helps here by showing type information right in the SDK.
```

Comment:
```suggestion
- **Strongly typed resources** – A tricky part about defining Kubernetes resources in YAML/JSON is the lack of typing. It can be hard to track down typos, and you have to apply the resource to a cluster to get any feedback. Pulumi helps here by showing type information right in the SDK.
```